### PR TITLE
Ignore modules with missing main source set

### LINKED ISSUE #11370: Gradle Plugin not working if module does not have main sourceSet
**Describe the bug**
Quarkus Version: 1.7.0.Final
Quarkus Gradle fails building the application because the plugin scans the workspace and retrieves the "main" sourceSet of all modules, but there is no "main" sourceSet in some modules.

For example: I have kotlin MPP submodules where there is a javaconvention but the javaconvention has no sourceSets.

**Expected behavior**
Should not fail.

**Actual behavior**
Build fails with: 
```
Caused by: org.gradle.api.UnknownDomainObjectException: SourceSet with name 'main' not found.
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection.createNotFoundException(DefaultNamedDomainObjectCollection.java:504)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection.getByName(DefaultNamedDomainObjectCollection.java:333)
        at io.quarkus.gradle.tasks.QuarkusGradleUtils.getSourceSet(QuarkusGradleUtils.java:41)
        at io.quarkus.gradle.builder.QuarkusModelBuilder.getWorkspaceModule(QuarkusModelBuilder.java:121)
        at io.quarkus.gradle.builder.QuarkusModelBuilder.getWorkspace(QuarkusModelBuilder.java:113)
        at io.quarkus.gradle.builder.QuarkusModelBuilder.buildAll(QuarkusModelBuilder.java:100)
        at io.quarkus.gradle.QuarkusPluginExtension.create(QuarkusPluginExtension.java:185)
        at io.quarkus.gradle.QuarkusPluginExtension.getQuarkusModel(QuarkusPluginExtension.java:178)
        at io.quarkus.gradle.QuarkusPluginExtension.getAppModelResolver(QuarkusPluginExtension.java:170)
        at io.quarkus.gradle.QuarkusPluginExtension.getAppModelResolver(QuarkusPluginExtension.java:166)
        at io.quarkus.gradle.QuarkusPluginExtension_Decorated.getAppModelResolver(Unknown Source)
        at io.quarkus.gradle.tasks.QuarkusBuild.buildQuarkus(QuarkusBuild.java:97)
...
```

**Suggestions for a fix:**
I have several ideas to fix this problem:
- Only scan modules that are related through dependencies to the module that is to be build.
- Ignore modules which have no "main" sourceSet
- Allow configuration of "to be excluded" sourceSets
- Allow configuration of the "main" sourceSet for each module

I think the second one is the easiest to implement, but i have no idea about the impact it would have.

### PR DESCRIPTION
**Pull request by antonwiens**
Fixes #11370

i added a module to the multi-module test gradle project where the main source set gets deleted.

The test "shouldLoadMultiModuleModel" verifies that the module does not get added to the workspace

I also verified the build is working with my own project where it did previously not.

### CODE REVIEW

**Comment by antonwiens** [comment]
Failed tests are unrelated to my changes, do i see that right?

File: devtools/gradle/src/test/resources/multi-module-project/settings.gradle [hunk 1]
```diff
@@ -16,4 +16,4 @@ pluginManagement {
 }
 rootProject.name='mutli-module-project'
 
-include 'common', 'application'
\ No newline at end of file
+include 'common', 'application', "nomain"
```
Comment (glefloch) [inline devtools/gradle/src/test/resources/multi-module-project/settings.gradle:19]: You can use single quotes here

**Review by glefloch — APPROVED** [review]
LGTM, just a little syntax comment.

**Comment by antonwiens** [comment]
Will it make it into 1.7.1?

**Review by glefloch — COMMENTED** [review]
LGTM, can you squash your commits? @aloubyansky WDYT? CI failures don't seem to be related.

**Comment by aloubyansky** [comment]
Let's wait for the current CI to finish.

**Review by aloubyansky — APPROVED** [review]

**Comment by aloubyansky** [comment]
Thanks a lot @antonwiens

**Pull request merged**